src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,41 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final int DATE_INDEX = 0;
+    private static final int NAME_INDEX = 1;
+    private static final int TIME_INDEX = 2;
+    private static final int PROFIT_INDEX = 3;
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder builder = new StringBuilder();
+        builder.append("Report for period ")
+                .append(dateFrom)
+                .append(" - ")
+                .append(dateTo);
+
+        for (String name : names) {
+            int salary = 0;
+            for (String content : data) {
+                String[] dataContent = content.split(" ");
+                LocalDate workDay = LocalDate.parse(dataContent[DATE_INDEX], FORMATTER);
+                LocalDate startDate = LocalDate.parse(dateFrom, FORMATTER);
+                LocalDate endDate = LocalDate.parse(dateTo, FORMATTER);
```

Comment:
This variables should be created before loops, cause they don't change when the application executes. You don't need to parse them every iteration